### ISSUE TITLE
crash because of missing project ID

### ISSUE THREAD

**Issue by NeuerUser** [kriztan/Pix-Art-Messenger#501]
#### General information

* **Version:** 2.5.1 (play store)
* **Device:** Xiaomi Mi 9
* **Android Version:**  Android 10(stock)
* **Server name:** conversations.im, jabber.de or self hosted
* **Server software:** ejabberd 19.09.1 or prosody 0.11.3 (if known)
* **Installed server modules:** Stream Managment, CSI, MAM
* **Pix-Art Messenger source:** PlayStore, PlayStore Beta Channel, F-Droid, Github, Codeberg, self build (latest HEAD)

java.lang.IllegalArgumentException: Please set your project ID. A valid Firebase project ID is required to communicate with Firebase server APIs: It identifies your project with Google.
	at com.google.android.gms.common.internal.Preconditions.checkNotEmpty(Unknown Source:8)
	at com.google.firebase.iid.FirebaseInstanceId.zza(com.google.firebase:firebase-iid@@20.2.2:52)
	at com.google.firebase.iid.FirebaseInstanceId.getInstance(com.google.firebase:firebase-iid@@20.2.2:2)
	at com.google.firebase.iid.FirebaseInstanceId.getInstance(com.google.firebase:firebase-iid@@20.2.2:1)
	at eu.siacs.conversations.services.PushManagementService.retrieveFcmInstanceToken(PushManagementService.java:123)
	at eu.siacs.conversations.services.PushManagementService.registerPushTokenOnServer(PushManagementService.java:42)
	at eu.siacs.conversations.services.XmppConnectionService$3.onBind(XmppConnectionService.java:349)
	at eu.siacs.conversations.xmpp.XmppConnection.finalizeBind(XmppConnection.java:1298)
	at eu.siacs.conversations.xmpp.XmppConnection.lambda$sendServiceDiscoveryInfo$5$XmppConnection(XmppConnection.java:1274)
	at eu.siacs.conversations.xmpp.-$$Lambda$XmppConnection$lIyM6Rh5oOIOcGbh4vscF90hlNI.onIqPacketReceived(Unknown Source:4)
	at eu.siacs.conversations.xmpp.XmppConnection.processIq(XmppConnection.java:782)
	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:658)
	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:502)
	at eu.siacs.conversations.xmpp.XmppConnection.switchOverToTls(XmppConnection.java:826)
	at eu.siacs.conversations.xmpp.XmppConnection.processStream(XmppConnection.java:486)
	at eu.siacs.conversations.xmpp.XmppConnection.connect(XmppConnection.java:388)
	at eu.siacs.conversations.xmpp.XmppConnection.run(XmppConnection.java:473)
	at java.lang.Thread.run(Thread.java:919)

#### Steps to reproduce

1. ?
2. ?

#### Expected result

What is the expected output? 

#### Actual result

What do you see instead?

#### Debug output

Please post the output of adb logcat. The log should begin with the start of Pix-Art Messenger and include all the
steps it takes to reproduce the problem.

````
Linux: adb -d logcat -v time | grep -i Pix-Art > logcat.txt
Windows: adb -d logcat -v time | FINDSTR Pix-Art > logcat.txt
````

**Comment by kriztan** [comment]
I need steps to reproduce

**Comment by NeuerUser** [comment]
Well, that's difficult, as there is nothing that I can do. I have been using pix-art for more than a year now without any issues. This morning the app was updated via play store to 2.5.1, which immediately lead to the error (happening even in the background). When I try to open the app, I get this error immediately. No steps to do.

I hoped the error message would be clear enough to identify the issue (and actually hoped that there was just a packaging error in the release).
If that is not the case, then I guess there is nothing I can do to help. I tried replacing the app with the last version, but the installer fails installing an older version on top of the current installed one.

I could try uninstalling and reinstalling it, but would lose all my settings and messages. :( Or is there any way to preserve this?

**Comment by kriztan** [comment]
Please wait... it seems to be an updated library from google. I'm going to publish an update today in the evening.

**Comment by FoxKyong** [comment]
I can confirm exactly the same problem after the update. I start Pix-art, it starts connecting to a server and then it crashes. Here is my crash log which Pix-art wants to send but can't because of crashes. Here is my crash log:

Device: asus ASUS_Z01RD
Android SDK: 29 (10)
Version: 2.5.1-playstore
Version: 2.5.1-playstore(305)
Last Update: 2020-07-17

java.lang.RuntimeException: Unable to start service eu.siacs.conversations.services.XmppConnectionService@e9f2e6e with Intent { act=eu.siacs.conversations.POST_CONNECTIVITY_CHANGE cmp=de.pixart.messenger/eu.siacs.conversations.services.XmppConnectionService (has extras) }: java.lang.IllegalArgumentException: Please set your project ID. A valid Firebase project ID is required to communicate with Firebase server APIs: It identifies your project with Google.
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4131)
	at android.app.ActivityThread.access$1800(ActivityThread.java:220)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1897)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:235)
	at android.app.ActivityThread.main(ActivityThread.java:7441)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: java.lang.IllegalArgumentException: Please set your project ID. A valid Firebase project ID is required to communicate with Firebase server APIs: It identifies your project with Google.
	at com.google.android.gms.common.internal.Preconditions.checkNotEmpty(Unknown Source:8)
	at com.google.firebase.iid.FirebaseInstanceId.zza(com.google.firebase:firebase-iid@@20.2.2:52)
	at com.google.firebase.iid.FirebaseInstanceId.getInstance(com.google.firebase:firebase-iid@@20.2.2:2)
	at com.google.firebase.iid.FirebaseInstanceId.getInstance(com.google.firebase:firebase-iid@@20.2.2:1)
	at eu.siacs.conversations.services.PushManagementService.retrieveFcmInstanceToken(PushManagementService.java:123)
	at eu.siacs.conversations.services.PushManagementService.registerPushTokenOnServer(PushManagementService.java:42)
	at eu.siacs.conversations.services.XmppConnectionService$3.onBind(XmppConnectionService.java:349)
	at eu.siacs.conversations.xmpp.XmppConnection.finalizeBind(XmppConnection.java:1298)
	at eu.siacs.conversations.xmpp.XmppConnection.sendDiscoTimeout(XmppConnection.java:1157)
	at eu.siacs.conversations.services.XmppConnectionService.processAccountState(XmppConnectionService.java:913)
	at eu.siacs.conversations.services.XmppConnectionService.onStartCommand(XmppConnectionService.java:794)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4113)
	... 8 more

**Comment by kriztan** [comment]
Thanks. I can confirm that there were changes in the firebase library. You can try installing the github version 2.5.1 which works without firebase (push messages) and update to 2.5.2 via playstore if it will be released. Both versions (github and playstore) are compatible to each other, but for automatically playstore updates you must once update from github version to playstore version manually. 
[https://github.com/kriztan/Pix-Art-Messenger/releases/download/2.5.1/PixArtMessenger-2.5.1-git-release.apk](url)

**Comment by kriztan** [comment]
Further information can be found here https://github.com/kriztan/Pix-Art-Messenger/issues/502

**Comment by NeuerUser** [comment]
Thanks a lot for the fast reaction and fix. Very much appreciated!

A small question that suppose doing my tries to install an older version: MIUI had a backup and restore function. However, it cannot backup or restore pix-art messenger (as well as many other apps). I think this is a flag set by the developer that determines whether the Android backup system (which MIUI is also using) is allowed to backup an app and it's data or not.

Is it true that this flag is providing pix-art messenger to be backed up? If yes, why was that done?

**Comment by kriztan** [comment]
It's true, since the messenger uses it's own backup/restore the global feature is disabled.

**Comment by NeuerUser** [comment]
> It's true, since the messenger uses it's own backup/restore the global feature is disabled.

Ah, I see. However, wouldn't it be possible to give the user the choice to decide on what backup/restore system he prefers to use?
For me personally, I prefer to automatically backup all (supported) apps at once, instead of going into each app individually and backup their data. I see that other users may think differently and prefer the manual in-built backup mechanism. But I would appreciate, if you would allow the users to freely choose which system they want to use.

Thanks again for the great messenger and for all your work.

**Comment by kriztan** [comment]
fixed via 700b7d14ccffe3a3fe9fecfa08feafd5efeb4016
